### PR TITLE
Bug fixes to Sort and updates to Sort UG

### LINKED ISSUE #157: [PE-D][Tester C] Sorting by attendance
When sorting by attendance for ascending and descending, it gives different results with those with no attendance record. For ascending, it puts those with no record at the bottom but for descending it put those with no record at the top.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: maxng17/ped#14

### LINKED ISSUE #158: [PE-D][Tester E] Sorting order is affected by casing of the attribute
When executing the commands `sort dsc a/phone` and `sort dsc a/PHONE`, one would expect the same sorting order to be outputted. However the app has a different output for each, as seen in the images below.

I would think the fix would be to make the a/ tag be non case sensitive, so that the correct and same order is used for both commands below.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: riccqi/ped#13

### LINKED ISSUE #159: [PE-D][Tester C] Error message of sort
The supported attributes listed here are different from the attributes listed on the user guide

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: maxng17/ped#15

### PR DESCRIPTION
**Pull request by highorbit25**
- Fix #159 and #160 to reflect the correct list of supported attributes that can be sorted in the error message.
- Fix #158 regarding the stability of the sort for duplicate attribute values when the same sort command is run multiple times by using `name` as a tie breaker.
- Fix #157 by ensuring those students with missing values for the specified sorting `Attribute` are always sorted to the bottom of the StudMap. This is done by returning the corresponding `Comparator` according to the specified sorting `Order`.
- Updates to UG to clarify usage of `Sort`

### CODE REVIEW

File: src/main/java/seedu/studmap/model/student/Student.java [hunk 1]
```diff
@@ -173,6 +201,30 @@ public float getAttendancePercentage() {
         return presentFor / numOfClasses * 100;
     }
 
+    /**
+     * Returns attendance in percentage for descending order sorting.
+     */
+    public float getAttendancePercentageForDsc() {
+        float numOfClasses = getAttendances().size();
+        if (numOfClasses == 0) {
+            return -Float.MIN_VALUE;
```
Comment (comicalromance) [inline src/main/java/seedu/studmap/model/student/Student.java:210]: This technically works, but Float.MIN_VALUE is actually a value very close to zero, so getting the smallest value would be -Float.MAX_VALUE instead

same for the other percentage comparators

**Review by comicalromance — APPROVED** [review]
LGTM!

File: src/main/java/seedu/studmap/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -38,7 +39,7 @@ public class SortCommand extends Command {
      */
     public SortCommand(Comparator<Student> comparator, String attributeType, Order sortOrder) {
         this.comparator = comparator;
-        this.attributeType = attributeType;
+        this.attributeType = attributeType.toLowerCase();
```
Comment (comicalromance) [inline src/main/java/seedu/studmap/logic/commands/SortCommand.java:42]: I'm not sure if this really solves #158, even running a sort with the exact same parameters can lead to different results

Comment (comicalromance) [reply src/main/java/seedu/studmap/logic/commands/SortCommand.java:42]: Found the issue, the sort uses studentID to break ties, but as studentID can sometimes be equal too the sort might be unstable, will add the additional tiebreak check to also compare using name then studentID

Comment (highorbit25) [reply src/main/java/seedu/studmap/logic/commands/SortCommand.java:42]: Yep! This was intended to solve the different command success message that comes out in upper case  "PHONE" when user specifies the attribute in upper case when typing the command. eg `sort asc a/PHONE`

**Pull request merged**